gprofiler/main.py

```diff
@@ -906,13 +926,28 @@ def setup_env(should_disable_core_files: bool, pid_file: str) -> None:
             logger.exception(f"Failed to write pid to '{pid_file}', continuing anyway")
 
 
+def pids_to_processes(args: configargparse.Namespace) -> Optional[List[Process]]:
+    if args.pids_to_profile is not None:
+        processes_to_profile = []
+        for pid in args.pids_to_profile:
+            try:
+                process = Process(pid)
+                processes_to_profile.append(process)
+            except NoSuchProcess:
+                continue
+        return processes_to_profile
```

Comment:
Let's add a log here, in the `is not None` branch - INFO level - of all PIDs that gProfiler will profile this run. Something like "Target PIDs given by --pids: 1,2,3,4".

Reply:
Added log here but I had to move `pids_to_processes()` and `verify_preconditions()` in `main()` to be called after initializing` logger`. I am not sure if that is ok? The other option is instead of logging it here we can add one more check after `logger` is initialized in `main()`

Reply:
Ok I see that one of the test from test_preconditions.py fails I guess its not ok then :D

Reply:
Changed to second option

Reply:
> The other option is instead of logging it here we can add one more check after logger is initialized in main() and log it there

That's indeed the correct choice. Looks good now.